verilog/analysis/symbol_table.cc

```diff
@@ -972,12 +1010,153 @@ class SymbolTable::Builder : public TreeContextVisitor {
                                                 SymbolMetaType metatype) {
     const auto p = current_scope_->TryEmplace(
         name, SymbolInfo{metatype, source_, &element});
-    if (!p.second) {
+    if (!p.second && p.first->second.Value().is_port_identifier) {
+      p.first->second.Value().supplement_definitions.push_back(name);
+    } else if (!p.second) {
       DiagnoseSymbolAlreadyExists(name, p.first->second);
     }
     return &p.first->second;  // scope of the new (or pre-existing symbol)
   }
 
+  // checks whether a given tag belongs to one of the listed tags
+  bool IsTagMatching(int tag, const std::vector<int>& tags) {
+    return std::find(tags.begin(), tags.end(), tag) != tags.end();
+  }
+
+  // checks if the current first leaf has conflicting information with the
+  // second symbol
+  bool IsTypeLeafConflicting(const SyntaxTreeLeaf* first,
+                             const verible::Symbol* second) {
+    if (!first || !second) return false;
+    if (IsTagMatching(second->Tag().tag,
+                      {static_cast<int>(NodeEnum::kPackedDimensions),
+                       static_cast<int>(NodeEnum::kUnpackedDimensions)})) {
+      return false;
+    }
+    if (second->Kind() == verible::SymbolKind::kLeaf) {
+      const SyntaxTreeLeaf* second_leaf =
+          verible::down_cast<const SyntaxTreeLeaf*>(second);
+      // conflict if there are multiple direction specifications
+      const std::vector<int> directiontags = {
+          verilog_tokentype::TK_input, verilog_tokentype::TK_output,
+          verilog_tokentype::TK_inout, verilog_tokentype::TK_ref};
+
+      bool is_first_direction = IsTagMatching(first->Tag().tag, directiontags);
+      bool is_second_direction =
+          IsTagMatching(second_leaf->Tag().tag, directiontags);
+
+      // conflict if there are multiple sign specifications
+      const std::vector<int> signtags = {verilog_tokentype::TK_signed,
+                                         verilog_tokentype::TK_unsigned};
```

Comment:
`const auto` will create `std::initializer_list<verilog_tokentype>`, which is be incompatible with `std::initializer_list<int>`. The `int` is used here, since later this `isTagMatching` method is reused for `NodeEnum`s